pkg/controller/operators/olm/envvar/initializer.go

```diff
@@ -58,14 +59,19 @@ func (d *DeploymentInitializer) initialize(ownerCSV ownerutil.Owner, deployment
 		}
 	}
 
-	merged = append(podConfigEnvVar, proxyEnvVar...)
+	// In pod configuration, an empty env var will allow us to set target to empty
+	// On the other hand, for proxy env vars, empty means that the env variable
+	// will not be set.
+	podConfigEnv := ToOverrideList(false, podConfigEnvVar...)
+	proxyEnv := ToOverrideList(true, proxyEnvVar...)
+	overrides = append(podConfigEnv, proxyEnv...)
```

Comment:
Can we safely remove the `Override` abstraction? along the lines of:

```go
func toMap(env []corev1.EnvVar) map[string]corev1.EnvVar {
	envMap := map[string]corev1.EnvVar{}
	for i := range env {
		e := env[i]
		envMap[e.Name] = e
	}
	return envMap
}

func overrideIgnoreEmpty(proxyEnv map[string]corev1.EnvVar, []corev1.EnvVar) {
  for _, e := proxyConfigEnv {
    if e.Value != "" {
        proxyEnv[e.Name] = e
     }
  }
}

merged := overrideIgnoreEmpty(toMap(proxyEnv), proxyConfigEnv)
```

Reply:
I was thinking of the scenario where these proxy env variables are already set on the deployment `PodSpec`. If we remove them ( if they are empty ) here, we will not have enough context to unset them from the target `PodSpec`. But I assume we will never have that scenario, correct? Our target `PodSpec` is always generated/calculated and never from an existing `PodSpec`. Also, we won't have users specif proxy env vars in `csv`, I assume. 

I can go back to what we had before and simply drop the empty proxy env vars here.

Reply:
> Our target PodSpec is always generated/calculated and never from an existing PodSpec

Yes, we're always calculating what the pod spec should be and then checking to ensure the on-cluster one matches.

> Also, we won't have users specify proxy env vars in csv, I assume.

It seems unlikely - but if so I would think that anything in the config would overwrite them?

Reply:
> but if so I would think that anything in the config would overwrite them?

Yes, anything in the subscription pod config will overwrite them. But not in the case where subscription pod config is `nil` and the `cluster` `Proxy` object has empty proxy env vars, then if we drop them here we lose this contextual information. 
I am assuming the order of precedence is, ( from high to low ) subscription pod config -> `cluster` proxy env var(s) -> `csv` spec.

To summarize, if we drop `UnsetIfEmpty` ( this is a package level abstraction not at the OLM API level ), this is what we have:
- Empty value in subscription pod `config` will overwrite target.
- We drop empty env var from `cluster` proxy object.
- We expect that the target `PodSpec` will never have proxy variables set.

If this sounds good, I will go ahead and remove  `UnsetIfEmpty`

Reply:
That sounds good to me